openedx_external_enrollments/tests/external_enrollments/tests_icc_external_enrollment.py

```diff
@@ -66,3 +73,248 @@ def test_str(self):
+            </SINGLE>
+        </RESPONSE>'''
+        response = Mock()
+        response.content = found_user_xml_response
+        expected_icc_user = {
+            'id': '1002',
+            'username': 'michael'
+        }
+
+        self.assertEqual(
+            self.base._get_icc_user_from_xml_response(response, 'get_user'),  # pylint: disable=protected-access
+            expected_icc_user
+        )
+
+    @patch.object(ICCExternalEnrollment, '_create_icc_user')
+    def test_validate_icc_user(self, create_icc_user_mock):
+        """
+        Testing _validate_icc_user method.
+        """
+        create_icc_user_mock.return_value = {
+            'id': '1002',
+            'username': 'michael'
+        }
+        expected_icc_user = create_icc_user_mock.return_value
+        self.assertEqual(
+            self.base._validate_icc_user({}, {}),  # pylint: disable=protected-access
+            expected_icc_user
+        )
+        create_icc_user_mock.assert_called_once_with({}, False)
+
+    @patch('openedx_external_enrollments.external_enrollments.icc_external_enrollment.get_user')
+    @patch('openedx_external_enrollments.external_enrollments.icc_external_enrollment.requests.post')
+    @patch('openedx_external_enrollments.external_enrollments.icc_external_enrollment.configuration_helpers')
+    @patch.object(ICCExternalEnrollment, '_get_random_string')
+    def test_create_icc_user(self, get_random_string_mock, configuration_helpers_mock, mock_post, get_user_mock):
+        """
+        Testing _create_icc_user method.
+        """
+        data = {
+            'user_email': 'test-email',
+        }
+        user_mock = Mock()
+        user_mock.username = 'user-test-username'
+        user_mock.email = 'user-test-email'
+        user_mock.first_name = 'user-test-firstname'
+        user_mock.last_name = 'user-test-lastname'
+        get_user_mock.return_value = (user_mock, Mock())
+        get_random_string_mock.return_value = 'test-password'
+        configuration_helpers_mock.get_value.return_value = 'test-auth-method'
+        mock_post.return_value.content = '''<?xml version="1.0" encoding="UTF-8" ?>
+        <RESPONSE>
+            <MULTIPLE>
+                <SINGLE>
+                    <KEY name="id">
+                        <VALUE>1223</VALUE>
+                    </KEY>
+                    <KEY name="username">
+                        <VALUE>testuser</VALUE>
+                    </KEY>
+                </SINGLE>
+            </MULTIPLE>
+        </RESPONSE>'''
+
+        expected_icc_user = {
+            'id': '1223',
+            'username': 'testuser',
+        }
+
+        self.assertEqual(
+            self.base._create_icc_user(data, False),  # pylint: disable=protected-access
+            expected_icc_user
+        )
+        get_random_string_mock.assert_called_once()
+        get_user_mock.assert_called_once_with(email=data.get('user_email'))
+
+        mock_post.side_effect = Exception('Test')
```

Comment:
This must be wrapped in a single test case. @ivanvgh

Reply:
Done.